security/providers/oidc/src/main/java/io/helidon/security/providers/oidc/OidcProvider.java

```diff
@@ -251,16 +263,19 @@ protected AuthenticationResponse syncAuthenticate(ProviderRequest providerReques
                 }
             }
         } catch (SecurityException e) {
+            LOGGER.log(Level.FINEST, "Failed to extract token from one of the configured locations", e);
             return failOrAbstain("Failed to extract one of the configured tokens" + e);
         }
 
         if (token.isPresent()) {
             return validateToken(providerRequest, token.get());
         } else {
-            return errorResponse(providerRequest,
-                                 Http.Status.UNAUTHORIZED_401,
-                                 null,
-                                 "Missing token, could not find in either of: " + missingLocations);
+            LOGGER.finest(() -> "Missing token, could not find in either of: " + missingLocations);
+            return CompletableFuture.completedFuture(errorResponse(providerRequest,
```

Comment:
I am not going to change this to Single in this PR. This would require a major change in whole security, which was created before our reactive module was in place. I am not sure - there may be a use where we have more than one `thenAccept` or `thenApply` on the same future, which would just fail with `Single`